### ISSUE TITLE
Refactor SNF to be `scikit-learn`-style

### ISSUE THREAD

**Issue by rmarkello** [rmarkello/snfpy#1]
SNF would perform pretty well as a data `transformer` with a `.fit_transform()` method, to which the raw `samples` x `features` arrays could be provided.

The [`scikit-learn` developer guidelines](http://scikit-learn.org/stable/developers/contributing.html#rolling-your-own-estimator) have some pretty extensive information on what this might look like, and provide a [project template](https://github.com/scikit-learn-contrib/project-template/) to help in implementation.

**Comment by rmarkello** [comment]
I thought about this a bit more and I don't think this style will necessarily work with SNF, given that the original data is required for propagating labels to new data. Closing for now.
